### PR TITLE
Deprecate Groups

### PR DESCRIPTION
**Pull request by XWB**
@stof Any more areas where we need to trigger a deprecation notice?

### CODE REVIEW

**Comment by stof** [comment]
please also add the `@deprecated` phpdoc so that IDEs and static analysers know about the deprecation.

**Comment by stof** [comment]
also, you need to tag some tests as `@legacy` when the cover the deprecated group feature (for configuration test, extract any group-related test to dedicated tests that can be legacy ones, so that tests using other features are not marked as legacy).

**Comment by stof** [comment]
`Remaining self deprecation notices (9)` tells me that your handling of tests is incomplete.

**Comment by rosier** [comment]
> @stof Any more areas where we need to trigger a deprecation notice?

To remove the dependency on `symfony/templating` maybe also deprecate `FOS\UserBundle\Mailer\Mailer` ?

**Comment by stof** [comment]
> To remove the dependency on `symfony/templating` maybe also deprecate `FOS\UserBundle\Mailer\Mailer`

yes, we should deprecate it as well (SwiftTwigMailer is a better one anyway). But please submit it as a separate PR than the group deprecation.

File: Model/GroupInterface.php [hunk 1]
```diff
@@ -11,9 +11,13 @@
 
 namespace FOS\UserBundle\Model;
 
+@trigger_error('Using Groups is deprecated since version 2.2 and will be removed in 3.0.', E_USER_DEPRECATED);
```
Comment (soyuka) [inline Model/GroupInterface.php:14]: this should only be in the deprecated class constructor as it triggers deprecation even if we don't use the group class

Comment (dunglas) [reply Model/GroupInterface.php:14]: It's more than that. The `GroupableInterface` is also deprecated, and implemented by the main `User` class...
I suggest to revert this PR, it doesn't look ready yet.

**Pull request merged**